kstreams-app/Dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM maven:3.6.1-jdk-8 AS build
```

Comment:
Is this Dockerfile valid? It references an old CP 5.4.0 version, and files like `pom.xml` which don't seem to exist.

Reply:
@mikebin it might be there as an archive?  @rspurgeon may know.

FYI, we are going to rewrite this application using Java Spring, so this PR is meant just as an intermediary step for migration of code into the repo.  So my leaning is to leave everything as-is.

Reply:
@ybyzek We could update this to use the CONFLUENT config.env parameter now that this is in cp-demo.

@mikebin I'm not sure what you mean "files like pom.xml" which don't seem to exist?

Reply:
@rspurgeon - I've probably just misread, but it looks like the Dockerfile is trying to copy a `pom.xml` to the image, but I don't see a `pom.xml` in the `kstreams-app/` folder - only `build.gradle`.

Reply:
@rspurgeon IIUC @mikebin is referring to this line: https://github.com/confluentinc/cp-demo/blob/DEVX-2138/kstreams-app/Dockerfile#L5

And I don't know the history, but my guess is that this application first had a pom.xml and relied on Maven, and then was converted to Gradle...but the reference to the pom file remained.

Reply:
:homer-doh:! yes of course.

I think I remember.  At first I used this mvn and this Dockerfile but then I switched to using Jib, so the dockerfile is built from with the gradle jib plugin.  I think this Dockerfile can be removed entirely.

Reply:
🥇 beauty.  I removed the Dockerfile and verified `make clean package` rebuilds it.